R/dependencies.R

```diff
@@ -53,36 +52,41 @@ dependency_table <- function(project = ".", feature = NULL,
   # deps$downstream_deps[["a"]] is a vector of downstream_deps of "a"
   # where the elements of the lists are the package names found in internal_deps
   # deps is ordered topologically
-  deps <- get_true_deps_graph(internal_deps, direction = c("upstream", "downstream"))
+  deps <- get_true_deps_graph(internal_deps, graph_directions = c("upstream", "downstream"))
 
-  current_repo <- internal_deps$package_name[internal_deps$repo == repo_deps_info$current_repo$repo &
-                                             internal_deps$host == repo_deps_info$current_repo$host]
+  current_pkg <- internal_deps$package_name[
+    internal_deps$repo == repo_deps_info$current_repo$repo &
+      internal_deps$host == repo_deps_info$current_repo$host
+    ]
 
-  internal_deps$type[internal_deps$package_name == current_repo] <- "current"
-  internal_deps$distance[internal_deps$package_name == current_repo] <- 0
+  internal_deps$type[internal_deps$package_name == current_pkg] <- "current"
+  internal_deps$distance[internal_deps$package_name == current_pkg] <- 0
 
-  upstream_nodes <- get_descendants_distance(deps[["upstream_deps"]], current_repo)
+  upstream_nodes <- get_descendants_distance(deps[["upstream_deps"]], current_pkg)
   internal_deps$type[internal_deps$package_name %in% upstream_nodes$id] <- "upstream"
-  internal_deps$distance[match(upstream_nodes$id, internal_deps$package_name)] <- upstream_nodes$distance
+  internal_deps$distance[internal_deps$package_name %in% upstream_nodes$id] <-
+    upstream_nodes$distance
 
-  downstream_nodes <- get_descendants_distance(deps[["downstream_deps"]], current_repo)
+  downstream_nodes <- get_descendants_distance(deps[["downstream_deps"]], current_pkg)
   internal_deps$type[internal_deps$package_name %in% downstream_nodes$id] <- "downstream"
-  internal_deps$distance[match(downstream_nodes$id, internal_deps$package_name)] <- downstream_nodes$distance
+  internal_deps$distance[internal_deps$package_name %in% downstream_nodes$id] <-
+    downstream_nodes$distance
 
   # sort the table
   internal_deps <- internal_deps[order(internal_deps$type, internal_deps$distance),
                                  c("package_name", "type", "distance", "branch",
                                    "repo", "host", "cache_dir")]
   rownames(internal_deps) <- NULL
-  internal_deps$type <- as.character(internal_deps$type)
+  # internal_deps$type <- as.character(internal_deps$type)
```

Comment:
Do we need this? Or is it ok for them to be factors

Reply:
`factor(c("a", "a", "b"), levels = c("a", "b")) == "a"` returns `[1]  TRUE  TRUE FALSE`, so it seems to work as expected.